packages/trading-widget/src/core-kit/hooks/trading/deposit-v2/use-deposit-quote.ts

```diff
@@ -0,0 +1,72 @@
+import BigNumber from 'bignumber.js'
+
+import { useEffect } from 'react'
+
+import {
+  useIsDepositTradingPanelType,
+  useReceiveTokenInput,
+  useSendTokenInput,
+  useTradingPanelPoolConfig,
+  useTradingPanelSettings,
+} from 'core-kit/hooks/state'
+import { useDebounce } from 'core-kit/hooks/utils'
+
+import { useDepositQuoteContractRead } from './use-deposit-quote-contract-read'
+import { useSwapDataBasedOnSendToken } from './use-swap-data-based-on-send-token'
+
+export const useDepositQuote = () => {
+  const { address, chainId } = useTradingPanelPoolConfig()
+  const isDeposit = useIsDepositTradingPanelType()
+
+  const [sendToken] = useSendTokenInput()
+  const [receiveToken, updateReceiveToken] = useReceiveTokenInput()
+  const [, updateSettings] = useTradingPanelSettings()
+  const debouncedSendTokenValue = useDebounce(sendToken.value, 500)
+  const { isFetching: isSwapDataFetching } = useSwapDataBasedOnSendToken()
+  const { data: quoteResult, isFetching: isDepositQuoteFetching } =
+    useDepositQuoteContractRead({
+      address,
+      chainId,
+    })
+
+  const isLoading =
+    isSwapDataFetching ||
+    isDepositQuoteFetching ||
+    sendToken.value !== debouncedSendTokenValue
+
+  useEffect(() => {
+    if (!isDeposit) {
```

Comment:
do we still have a use case of that hook out of deposit tab?

Reply:
it is still can be called even if withdraw tab is opened, I believe thats how @headlessui tabs work